pkg/util/detector/detector.go

```diff
@@ -221,13 +229,26 @@ func (d *ResourceDetector) EventFilter(obj interface{}) bool {
 		strings.HasPrefix(clusterWideKey.Namespace, names.KarmadaReservedNSPrefix) {
 		return false
 	}
-
 	if clusterWideKey.Group == clusterv1alpha1.GroupName ||
 		clusterWideKey.Group == policyv1alpha1.GroupName ||
 		clusterWideKey.Group == workv1alpha1.GroupName {
 		return false
 	}
 
+	if names.IgnoreKinds.Has(clusterWideKey.Kind) {
+		return false
+	}
+	watchedKinds := strings.Split(d.WatchedKinds, ",")
+	if len(watchedKinds) > 0 {
+		kindsSet := sets.NewString(watchedKinds...)
+		return kindsSet.Has(clusterWideKey.Kind)
+	}
+
+	watchedNamespaces := strings.Split(d.WatchedNamespaces, ",")
```

Comment:
It seems like logic here won't be executed if watchedKinds is set?

Reply:
good catch.If both watchedKinds and watchedNamespaces, it should only watch watchedKinds in watchedNamespaces.